backend/src/routes/InvitationRouter.ts

```diff
@@ -0,0 +1,213 @@
+import express, { Request } from "express";
+import { PrismaClient, Status } from "@prisma/client";
+import { StatusCodes } from "http-status-codes";
+import { TripParams } from "./TripRouter";
+
+const router = express.Router({ mergeParams: true });
+const prisma = new PrismaClient();
+
+export interface InvitationParams extends TripParams {
+  invitationId: string,
+  code?: string;
+  status?: string;
+}
+
+// temporary for testing until auth done
+const userID = "6669267e34f4cab1d9ddd751";
+
+// get all invitations
+router.get("/", async (req: Request, res) => {
+  try {
+    const trips = await prisma.tripMember.findMany();
+    console.log("trips", trips);
+    res.status(StatusCodes.OK).json(trips);
+  } catch (error) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while getting invitatiion." });
+  }
+});
+
+// get all received invitations of a user, including trip details
+router.get("/all-received", async (req: Request, res) => {
+  try {
+    const trips = await prisma.tripMember.findMany({
+      where: {
+        inviteeId: userID,
+        status: 'PENDING'
+      },
+      include: {
+        trip: true,
+        inviter: true
+      }
+    });
+
+    console.log("trips", trips);
+    res.status(StatusCodes.OK).json(trips);
+  } catch (error) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while getting invitatiion." });
+  }
+});
+
+// get all sent invitations of a user, including trip details
+router.get("/all-sent", async (req: Request, res) => {
+  try {
+    const trips = await prisma.tripMember.findMany({
+      where: {
+        inviterId: userID,
+      },
+      include: {
+        trip: true,
+        invitee: true,
+      }
+    });
+
+    if (!trips) {
+      res.status(StatusCodes.NOT_FOUND).json({ error: "No invitations found." });
+    }
+
+    res.status(StatusCodes.OK).json(trips);
+  } catch (error) {
+    console.error("Error retrieving sent invitations:", error);
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while getting invitation." });
+  }
+});
+
+// get an invitation by status
+// example endpoint: /invite?status=PENDING
+router.get("/", async (req: Request<InvitationParams>, res) => {
+  const { status } = req.query;
+  try {
+    const trips = await prisma.tripMember.findMany({
+      where: {
+        inviteeId: userID,
+        status: status as Status
+      },
+    });
+
+    res.status(StatusCodes.OK).json(trips);
+  } catch (error) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while getting invitatiion." });
+  }
+});
+
```

Comment:
You have two `GET` routes with the same path. This second route handler will never be reached because the first route to get all invitations will always match first, even if there is status. 

Consider combining the 2 routes and check for status, or use another path for this one.

Reply:
Great pointt!